Improving documentation for getting started with IDEA.

uglycoyote opened the pull request:
I wanted to improve this doc based on my not-so-great experience of trying to follow it.

Firstly, after trying things the hard way first, and then later discovering the IDEA Minecraft plugin, I felt that the latter was the way to go for a first time user.  I struggled with following the Gradle directions, which was wasted time when the plugin would have generated a working starting point much faster.   Unfortunately, the plugin instructions were buried way down the page, so I missed them.  So I wanted to improve those by making them nearer to the top. 

Secondly, assuming the user does not wish to use the IDEA Minecraft plugin, they may run in to the same issue that I did where the default settings for creating a Gradle project in IDEA don't enable the "create content roots" setting, making it less than obvious where to plugin code.   (After trying to create a plugin following this approach,  I failed to guess the correct location for my main java file and so my code was not being compiled.   I went on the discord channel for help and was told to go back and learn Java, and generally dismissed)

Thirdly, it was unclear to me where to go once the project was created.  Assuming I had a plugin project and was able to compile it with a `onServerStarted` event handler, what next?   As far as I could tell none of the docs gave any indication of how to run the plugin.  Perhaps this is also obvious to most members of the community, but as someone getting started this was not clear to me, so I have added a short section for this.  

I also changed some headings here to make clearer the distinction of using a template generated from the plugin vs. starting "from scratch"

uglycoyote committed:
Improving documentation for getting started with IDEA.

uglycoyote committed:
fixing underlines that were too short -- failing travis check

uglycoyote committed:
more underlines were too short, fixedem

Inscrutable commented on source/plugin/workspace/idea.rst:
```diff
@@ -5,11 +5,50 @@ Setting Up IntelliJ IDEA
 
-Gradle
-======
+If you want to create your project completely from scratch, please skip ahead to the Gradle or Maven sections.  Using the Minecraft Dev plugin sets up a working starting point and eliminates some of the guesswork in getting your project off the ground.
```
Please try to keep the lines 120 characters or less. You have a few very long lines here. It's one of our Docs Guidelines (which can be found on the [SpongeDocs](https://docs.spongepowered.org/stable/en/contributing/spongedocs.html), for reference).

Inscrutable commented on source/plugin/workspace/idea.rst:
```diff
@@ -5,11 +5,50 @@ Setting Up IntelliJ IDEA
+* Specify your desired plugin **description**, **authors**, **website**, and **dependencies** if you want.
+* Click ``Next`` to move on.
+* Verify your project name, location, and module information, then click ``Finish``.
+* The plugin will create a main java file as a starting point, with the logger already injected.  Try adding a ``logger.info`` statement in the ``onServerStart`` function to verify that the plugin is working when you run it.
```
If it's optional, then I suggest changing it to "You may add a ``logger.info`` "...

vectrixdevelops replied:
I would also refer to the onServerStart as an event instead of a function.

uglycoyote replied:
@connorhartley an "event handler" perhaps?  I would consider the "event" to be the thing that minecraft is sending, and onServerStart to be a "listener" or "event handler" function.

Inscrutable reviewed (changes requested):
An excellent contribution that also adds reference to the Minecraft Dev Plugin. Several over-long lines need breaking up (I'll clean up the editing later if I have to). The rest is great, although I want someone more familiar with the dev environment to review it too.

12awsomeman34 commented on source/plugin/workspace/idea.rst:
```diff
@@ -65,6 +103,20 @@ Creating Your Project
 * Enter your project's name, and click ``Finish``.
 
+
+Testing your plugin
```
Title case

12awsomeman34 commented on source/plugin/workspace/idea.rst:
```diff
@@ -5,11 +5,50 @@ Setting Up IntelliJ IDEA
+Refer to the Gradle_ or Maven_ configuration sections, depending on what you chose during project creation.
+
+
+
```
Why the extra new lines between sections in some parts of this page?

uglycoyote replied:
@NariKiro Two lines before a header look make things look less crammed together (not sure why I used 3 here) but since one line before headers seems to be the standard, I can ditch the extra spaces.

12awsomeman34 replied:
You're welcome to suggest a change in the standard if you want, but the extra lines only really fix readability of the source file, which most people don't look at.

uglycoyote committed:
changes from pull request review

uglycoyote commented:
I wrapped the long lines, got rid of the extra spaces, and incorporated some of the wording changes suggested above.

Spongy commented:
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/af52884/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/af52884/plugin/workspace/idea.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/725#issuecomment-408176343), not to the direct url to the preview._

ST-DDT commented on source/plugin/workspace/idea.rst:
```diff
@@ -5,11 +5,57 @@ Setting Up IntelliJ IDEA
+* Click ``Next`` to move on.
+* Verify your project name, location, and module information, then click ``Finish``.
+* The plugin will create a main java file as a starting point, with the logger already injected.
+  You may add a ``logger.info`` statement in the ``onServerStart`` event handler to verify that the plugin
```
Maybe add some brackets to `logger.info()` to make it clearer that this refers to a method call.

ST-DDT reviewed (approved):
One small change remaining, otherwise it looks good to me.

uglycoyote committed:
Moving the 'Testing your Plugin' section

uglycoyote commented:
Reading it again, I realized that I had accidentally put the "Testing Your Project" section as a subheading under the Maven section, but the actual instructions were for Gradle.  I fixed this by moving "Testing Your Project" up to being a top-level section, and including information relevant to both Gradle and Maven.

DISCLAIMER: I haven't used Maven so I'm not sure if what I found is the best method for making a jar,  I just made a dummy project and walked through the process, which is similar to but not the same as in Gradle, so feel free to suggest something different.

uglycoyote committed:
Adding parens after logger.info

ryantheleach commented on source/plugin/workspace/idea.rst:
```diff
@@ -5,11 +5,50 @@ Setting Up IntelliJ IDEA
+
+  * Your **Group ID** should usually correspond to your Java package name. See :doc:`../plugin-class` for details.
+  * Your **Artifact ID** should usually correspond to your **plugin ID** you chose earlier, e.g. ``myplugin``.
+  * Your **Version** is up to you.
```
Your **Version** is up to you, but is typically in the format of major.minor.patch e.g. (1.0.0) so that's it's compatible with dependency ranges / other mod dependencies etc.

I attempted (and failed) to find a hard definition for what works for plugin dependencies, or forge dependencies, but I'm going to need someone with more knowledge to chime in here.

Inscrutable replied:
Is this nitpick all that is holding up this PR? We can always clarify it later if we can get a better definition.

uglycoyote replied:
If it matters, that stuff about group, artifact, and version are not my new text, just showing up as green in the diff because I moved paragraphs around, sorry for confusing matters that way.

So since that explanation of the version ID is already the status quo I don't see any reason to hold up this PR.  But I do agree that some additional guidance about appropriate version numbers would help (maybe could be added as a separate issue)

uglycoyote replied:
Added #727

Inscrutable commented:
This PR will also make some of @parlough 's monster WIP PR #691 redundant so It may be worth having a brief look at the IDEA page on that too @uglycoyote

uglycoyote commented:
@Inscrutable:  Were you thinking of the "Minecraft Dev Plugin" instructions?   Looking at the diffs it is a bit confusing who's adding what, but I think it's mostly the same content.  I did not add much to that section.. 

I took what was already there (from #703), moved it up higher in the file, and added the small section " Installing the Minecraft Dev Plugin" (previously it only told how to use, but not install)

looking at the diffs on #691, (and by his PR number) I'm guessing that @parlough started earlier and merged in the changes from #703.
